Tutorials/Notebooks/netket-VMC.ipynb

```diff
@@ -0,0 +1,426 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Variational Monte Carlo with Neural Networks\n",
+    "\n",
+    "In this tutorial we will use the NetKet software package to find the ground state of the J1-J2 model in one-dimension with periodic boundary condition. The Hamiltonian of the model is given by:\n",
+    "\n",
+    "$$ H = \\sum_{i=1}^{L} J_{1}\\hat{S}_{i} \\cdot \\hat{S}_{i+1} + J_{2} \\hat{S}_{i} \\cdot \\hat{S}_{i+2} $$\n",
+    "\n",
+    "where the sum is over sites of the 1-D chain.\n",
+    "\n",
+    "\n",
+    "We will also explore some useful functionalities provided by the package.\n",
+    "\n",
+    "## Objectives:\n",
+    "    1. Defining custom Hamiltonians\n",
+    "    2. Defining the machine (variational ansatz)\n",
+    "    3. Variational Monte Carlo Optimisation\n",
+    "    4. Measuring observables\n",
+    "    5. Data Visualisation\n",
+    "    6. Sanity Check: Exact Diagonalisation\n",
+    "    \n",
+    "Let's start."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import netket library\n",
+    "import netket as nk\n",
+    "from mpi4py import MPI\n",
+    "\n",
+    "# Helper libraries\n",
+    "import numpy as np\n",
+    "import networkx as nx\n",
+    "import matplotlib.pyplot as plt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1) Defining a Custom Hamiltonian\n",
+    "\n",
+    "The first thing to do is to define the graph (lattice) on which to specify the Hamiltonian. Here we would like to build a one-dimensional graph with both nearest and next nearest neighbour bonds. For this purpose, we use the ``networkx`` library."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Couplings J1 and J2\n",
+    "J = [1, 0.2]\n",
+    "L = 16"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define custom graph\n",
+    "G = nx.Graph()\n",
```

Comment:
Can we do this without using iGraph?

Reply:
But this is networkx

Reply:
I meant, just using a NetKet graph and not an external library to do this... I don't know if that's ugly or it can be done nicely with a CustomGraph (I guess it's fine, since this is just a 1D lattice?)

Reply:
ok its removed